packages/teko/src/Teko_Utilities.cpp

```diff
@@ -234,21 +234,21 @@ RCP<Tpetra::CrsMatrix<ST,LO,GO,NT> > buildGraphLaplacian(int dim,ST * coords,con
                                                        stencil.getGlobalMaxNumRowEntries()+1));
 
    // allocate an additional value for the diagonal, if neccessary
-   std::vector<ST> rowData(stencil.getGlobalMaxNumRowEntries()+1);
-   std::vector<GO> rowInd(stencil.getGlobalMaxNumRowEntries()+1);
+   auto sz = stencil.getGlobalMaxNumRowEntries()+1; 
+   Teuchos::Array<ST> rowData(sz);
+   Teuchos::Array<GO> rowInd(sz);
 
    // loop over all the rows
    for(LO j=0;j< (LO) gl->getNodeNumRows();j++) {
       GO row = gl->getRowMap()->getGlobalElement(j);
       ST diagValue = 0.0;
-      GO diagInd = -1;
+      int diagInd = -1;
```

Comment:
This could cause 64-bit - to - 32-bit assignment issues.  It would be better to do this:
```c++
GO diagInd = Teuchos::OrdinalTraits<GO>::invalid();
```

Reply:
I had that originally.  The problem is with the change, some of the Teko tests fail if I have ```Tpetra_INST_INT_UNSIGNED_LONG:BOOL=ON``` b/c there is logic assuming diagInd = -1, which it cannot be if the type of it is unsigned long.  I agree with you that with my change, there's a concern about a 64-bit -to - 32-bit assignment.  I think the question is, do we care if some Teko tests fail if ```Tpetra_INST_INT_UNSIGNED_LONG:BOOL=ON```  (as some Tpetra tests do).  If we are OK with them failing, we can keep the logic as it was originally.

Reply:
@ikalash I don't think Albany uses Teko, does it?  It might be enough just to fix Tpetra.

Reply:
@mhoemmen : it does use Teko actually, in the LCM package.  This usage will be expanded in the near future, as we switch the entire code to block data structures.  @eric-c-cyr seems to imply in his email that the changes should be OK, but I'd like him to confirm.  Also it appears auto-tester got relaunched, and I'm also doing a build now to reproduce one of the auto-tester builds - this might shed some light into whether or not there are potential issues.  I think the auto-tester build failed earlier b/c I accidentally indicated that I want to merge the PR into master, not develop.

Reply:
@mhoemmen : I just pushed a fix that should avoid Teko test failures and potential issues with 64-bit - to -32-bit conversions in the Laplacian graph routine (which @eric-c-cyr was concerned about) using ifdefs.  Let me know if you don't like the fix.

Reply:
I just pushed a better fix involving invalid() and no ifdefs.  I think we can say the issue has been resolved.  There are no test failures with GO=unsigned long.